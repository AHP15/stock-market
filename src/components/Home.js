import { useSelector } from 'react-redux';

import SymbolCard from './SymbolCard';

import styles from '../styles/Home.module.css';

const Home = () => {
  const { symbols, loading } = useSelector((state) => state.symbols);

  if (loading) {
    return <div>Loading...</div>;
  }
  return (
    <div className={styles.home}>
      {
        symbols.map((obj, i) => (
          <SymbolCard
            key={obj.symbol}
            symbol={obj.symbol}
            company={obj.companyName}
            price={obj.price}
            dark={i % 2 !== 0}
            index={i}
          />
        ))
      }
    </div>
  );
};

export default Home;
