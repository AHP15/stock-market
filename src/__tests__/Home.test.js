import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import '@testing-library/jest-dom';
import { BrowserRouter } from 'react-router-dom';

import Home from '../components/Home';

import Store from '../slices/store';

it('renders Details correctly', () => {
  const tree = renderer.create(
    <BrowserRouter>
      <Provider store={Store}><Home /></Provider>
    </BrowserRouter>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
