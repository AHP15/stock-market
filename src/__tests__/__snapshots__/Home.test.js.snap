// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Details correctly 1`] = `
<div
  className="home"
/>
`;
